src/Grpc.AspNetCore.Server/Internal/IGrpcServiceInvokerFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.Core;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    /// <summary>
+    /// An interface for creating delegates used to invoke service methods on .NET types.
+    /// </summary>
+    /// <typeparam name="TService">The service type.</typeparam>
+    internal interface IGrpcServiceInvokerFactory<TService>
+    {
+        UnaryServerMethod<TRequest, TResponse, TService> CreateUnaryInvoker<TRequest, TResponse>(Method<TRequest, TResponse> method);
```

Comment:
While in the default MapGrpcService() use case the TService is actually the type of the generated service class, 
if one uses the MapGrpcService() overload with binder options and custom IGrpcServiceInvokerFactory,  it is no longer clear that TService denotes a service class. It could be something else, depending on given IGrpcServiceInvokerFactory implementation?

Reply:
We need TService because that is part of the delegate signature, and it is the type that will be passed to the delegate when it is executed.

But TService doesn't necessarily have to be used by the delegate. In the inline tests it is ignored.

Does that make sense?

Reply:
Sure, I was just wondering if it should really be called TService when it can in fact represent other things too.

Reply:
Maybe. IMO we shouldn't think too hard about it because it is internal, and only in unit tests could it mean something else.